surfactantplugin-syft/surfactantplugin_syft.py

```diff
@@ -0,0 +1,98 @@
+import json
+import subprocess
+import time
+from typing import List, Optional
+
+import surfactant.plugin
+from surfactant.plugin.manager import get_plugin_manager
+from surfactant.sbomtypes import SBOM, Relationship, Software
+
+
+@surfactant.plugin.hookimpl
+def extract_child_info(
+    sbom: SBOM, software: Software, filename: str, filetype: str
+) -> Optional[List[Software]]:
+    pm = get_plugin_manager()
+    # Change to properly filter filetypes, add to if statement for filetypes syft should run for
+    if filetype == "TAR":
+        data = subprocess.check_output(
+            "anchore_syft " + filename + " -o json --scope all-layers", shell=True
+        )
+        data = json.loads(data.decode())
+        sw_list = []
+        for i in data["artifacts"]:
+            sw_entry = Software(
+                sha1=None,
+                # Syft does not provide a SHA256 for each artifact.
+                # This uses their unique IDs in place since surfactant is dependent on this hash.
+                sha256=i["id"],
```

Comment:
Files from within a tar file are a bit of an odd case. They should be able to get their own software entry rather than be a component; but Syft doesn't provide a hash for the files in it, which is awkward. For this let's add a TODO for me to discuss with Lucas. Or maybe better, open a GitHub issue rather than adding it as a comment.

```suggestion
                # Syft does not provide a SHA256 for each artifact.
                # This uses their unique IDs in place since surfactant is dependent on this hash.
                # TODO (Ryan) discuss how to handle software entry without a hash with Lucas in the CyTRICS SBOM schema
                sha256=i["id"],
```